05-Linking-Accounts/app/assets/javascripts/home.js.erb

```diff
@@ -0,0 +1,33 @@
+function signin() {
+	var options = {
+	   auth: {
+	 		redirectUrl: '<%= Rails.application.secrets.auth0_callback_url %>',
+	 		params: {
+	 			scope: 'openid name email picture'
+	 		}
+	   }
+	 };
+	var lock = new Auth0Lock('<%= Rails.application.secrets.auth0_client_id %>', '<%= Rails.application.secrets.auth0_domain %>', options);
+
+ 	lock.show();
+}
+
+function linkPasswordAccount(){
+	connection = $("#link_provider").val();
+  var linkOptions = {
+		auth: {
+			redirectUrl: '<%= Rails.application.secrets.auth0_callback_url %>',
+    	allowLogin: true
+		},
+    languageDictionary: {
+    	title: 'Link another account'
+    }
+  };
+  if (connection){
+    linkOptions.allowedConnections = [connection];
+  }
+	var linkLock = new Auth0Lock('<%= Rails.application.secrets.auth0_client_id %>', '<%= Rails.application.secrets.auth0_domain %>', linkOptions);
+
+  //open lock in signin mode, with the customized options for linking
+  linkLock.show();
```

Comment:
@falvarisqueta: not really, we need to set lock in such a way that it "forgets" the previous login, but the linkage occurs in the settings controller using the SDK (see line 97 of settings controller)